core/src/stack_size.rs

```diff
@@ -0,0 +1,19 @@
+//! Executables must specify their stack size by using the `stack_size!` macro.
+//! It takes a single argument, the desired stack size in bytes. Example:
+//! ```
+//! stack_size!{0x400}
+//! ```
+
+// stack_size works by putting a symbol equal to the size of the stack in the
+// .stack_buffer section. The linker script uses the .stack_buffer section to
+// size the stack. flash.sh looks for the symbol by name (hence #[no_mangle]) to
+// determine the size of the stack to pass to elf2tab.
+
+#[macro_export]
+macro_rules! stack_size {
```

Comment:
Is there any check for the anlignment of the stack? Is a check necessary?

Reply:
There is still an alignment check in the linker file (generic_layout.ld)

Reply:
I would count that as a "no".

Reply:
Check is the wrong word in this context: the linker _controls_ layout/alignment. There is an `ALIGN` directive in the layout file which creates a `_stack_top_aligned` symbol (currently aligned to 8 bytes, which is sufficient for all supported tock archs).

Reply:
That's not the point I wanted to make: My point is that if we do something in the Rust code which obviously can't work we should fail early with a meaningful error message. Maybe I'm just not smart enough, but the last time I solved alignment issues took me a day because I didn't understand how buffers for IPC have to be aligned - although I had errors - but the errors were not meaningful (to me).

Reply:
Ah, yeah. Strongly agreed with fail-fast. In this case I think it _should_ be fine, assuming that the libtock startup routines use the `_stack_top_aligned` symbol (which they should).

Reply:
I think several of us forgot that `layout_generic.ld` ends with an `ASSERT`. If I try to set a stack size that is not a multiple of 8 bytes, I get this error:

```
   Compiling libtock v0.2.0 (/home/jrvanwhy/libtock-rs)
    Blocking waiting for file lock on build directory==============>   ] 38/40: libtock                                                
error: linking with `rust-lld` failed: exit code: 1
  |
  = note: "rust-lld" "-flavor" "gnu" "-L" "/home/jrvanwhy/.rustup/toolchains/nightly-2020-06-10-x86_64-unknown-linux-gnu/lib/rustlib/riscv32imac-unknown-none-elf/lib" "/home/jrvanwhy/libtock-rs/target/riscv32imac-unknown-none-elf/release/examples/libtock_test-b7d5f39cd8a73110.libtock_test.eiijs1ym-cgu.0.rcgu.o" "-o" "/home/jrvanwhy/libtock-rs/target/riscv32imac-unknown-none-elf/release/examples/libtock_test-b7d5f39cd8a73110" "--gc-sections" "-L" "/home/jrvanwhy/libtock-rs/target/riscv32imac-unknown-none-elf/release/deps" "-L" "/home/jrvanwhy/libtock-rs/target/release/deps" "-L" "/home/jrvanwhy/.rustup/toolchains/nightly-2020-06-10-x86_64-unknown-linux-gnu/lib/rustlib/riscv32imac-unknown-none-elf/lib" "--start-group" "--end-group" "-Bstatic" "/home/jrvanwhy/.rustup/toolchains/nightly-2020-06-10-x86_64-unknown-linux-gnu/lib/rustlib/riscv32imac-unknown-none-elf/lib/libcompiler_builtins-f6d81fb1499935c8.rlib" "-Tlayout.ld" "-Bdynamic"
  = note: rust-lld: error: 
          STACK_SIZE must be a multiple of 8 bytes
          
          rust-lld: error: 
          STACK_SIZE must be a multiple of 8 bytes
          

error: aborting due to previous error

error: could not compile `libtock`.
```

I can reword it for clarity, but that seems clear enough to me.

Reply:
It is the same here